Sources/TuistCore/Utils/OtoolController.swift

```diff
@@ -0,0 +1,89 @@
+    func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]>
+}
+
+/// OtoolController
+/// Used to find out against which dynamic libraries a certain binary is linked.
+public struct OtoolController: OtoolControlling {
+    private let system: Systeming
+
+    public init(system: Systeming = System.shared) {
+        self.system = system
+    }
+
+    public func dlybDependenciesPath(forBinaryAt path: AbsolutePath) throws -> Single<[AbsolutePath]> {
+        let arguments = ["otool", "-L", path.pathString]
+
+        return System.shared.observable(arguments)
+            .mapToString()
+            .collectAndMergeOutput()
+            .map { try OtoolOutput(filePath: path, output: $0) }
+            .map { $0.paths }
+            .asSingle()
+    }
+
+    /// Given a framework path it returns the path to its binary.
+    /// - Parameter frameworkPath: Framework path.
+    func binaryPath(frameworkPath: AbsolutePath) -> AbsolutePath {
+        let frameworkName = frameworkPath.basename.replacingOccurrences(of: ".framework", with: "")
+        return frameworkPath.appending(component: frameworkName)
+    }
+}
+
+/// Otool output example:
+/// After running `otool -L path/to/AlamofireImage.framework/AlamofireImage` you will see something like this:
+///
+/// path/to/AlamofireImage.framework/AlamofireImage:
+///    @rpath/AlamofireImage.framework/AlamofireImage (compatibility version 1.0.0, current version 1.0.0)
+///    @rpath/Alamofire.framework/Alamofire (compatibility version 1.0.0, current version 1.0.0)
+///    /usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
+///    /System/Library/Frameworks/UIKit.framework/UIKit (compatibility version 1.0.0, current version 3987.9.0)
+///    @rpath/libswiftCore.dylib (compatibility version 1.0.0, current version 1200.2.26)
+///
+
+private struct OtoolOutput {
+    let paths: [AbsolutePath]
+
+    init(filePath: AbsolutePath, output: String) throws {
+        let outputByNewLines = Array(output.components(separatedBy: .newlines))
+        // first line contains an ':'at the end
+        guard outputByNewLines.first == "\(filePath.pathString):" else { throw OtoolErrors.invalidOutput }
+
+        // we need to remove the /FrameworkName.framework/FrameworkName
+        let folderPath = filePath.removingLastComponent().removingLastComponent()
+
+        paths = outputByNewLines
+            .dropFirst() // first line is the path to the framework
+            .dropLast() // last is an empty line
+            .filter { $0.contains("@rpath") } // the rest are system libraries
+            .filter { !$0.contains("dylib") } // swift libs
+            .map { $0.replacingOccurrences(of: "@rpath", with: folderPath.pathString) }
+            .compactMap { $0.split(separator: " ").first } // drop compatibility
+            .map { AbsolutePath($0.trimmingCharacters(in: .whitespacesAndNewlines)) }
+    }
+}
```

Comment:
@pepibumur was this what you had in mind?

Reply:
I can see that you had to add `RxSwift` and  `RxBlocking` to `TuistCore`'s dependencies. Im thinking if we can avoid it. 🤔

Reply:
Yes! That's exactly what we need to build an in-memory graph for Carthage frameworks.